lib/resque/scheduler/scheduling_extensions.rb

```diff
@@ -44,6 +44,16 @@ module SchedulingExtensions
       # param, otherwise params is passed in as the only parameter to
       # perform.
       def schedule=(schedule_hash)
+        if supports_lua?
+          setup_schedule(schedule_hash)
+        else
+          Resque.logger.warn!('setup_schedule! can cause a race condition, it is ' \
```

Comment:
I would move the warning to the method itself so that it isn't called elsewhere in the future without a warning.

The name could also be clearer that it shouldn't be used at the call-site, e.g. by naming the method `setup_schedule_non_atomically`.  Just using a `!` suffix doesn't imply much in the existing code (e.g. `log!` just means an info log message compared to `log` which is for debug logging.  There are also several bang methods without a non-bang method in the existing code.